File: easy/28_Find_the_Index_of_the_First_Occurrence_in_a_String/28_Find_the_Index_of_the_First_Occurrence_in_a_String.c
int strStr(char* haystack, char* needle) 
{
    if (strlen(haystack) < strlen(needle))
        return -1;

    int i;
    int k;
    int strstr;
    int ii;
    
    i = 0;
    while (haystack[i])
    {
        k = 0;
        strstr = i;
        ii = i;
        if (haystack[ii] == needle[k])
        {
            while (i < strlen(haystack) && k < strlen(needle) && haystack[ii] == needle[k])
            {
                ii++;
                k++;
            }
            if (needle[k] == '\0')
                return strstr;
        }
        i++;
    }
    return (-1);
}

File: easy/28_Find_the_Index_of_the_First_Occurrence_in_a_String/28_Find_the_Index_of_the_First_Occurrence_in_a_String.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        for i in range(len(haystack)):
            if haystack[i] == needle[0]:
                k = 0
                now = i
                while k < len(needle) and i < len(haystack) and haystack[i] == needle[k]:
                    i+=1
                    k+=1
                if k == len(needle):
                    return now
        return -1


test = Solution()
print(test.strStr("leetcode", "leeto"))